documentation/source/examples.rst

```diff
@@ -56,6 +56,8 @@ Other Examples
     examples/magnetostatics/helmholtz_example
     examples/materials/material_definitions
     examples/mesh/mesh_tutorial
+    examples/optimisation/geometry_optimisation
+    examples/optimisation/nonlinearly_constrained_problem
```

Comment:
We may want to consider putting one of these or a different one in introductory examples